kolibri/plugins/user/assets/src/views/sign-in-page/index.vue

```diff
@@ -68,30 +93,107 @@
       'icon-button': require('kolibri.coreVue.components.iconButton'),
       'core-textbox': require('kolibri.coreVue.components.textbox'),
       'logo': require('kolibri.coreVue.components.logo'),
+      'ui-autocomplete-suggestion': require('keen-ui/src/UiAutocompleteSuggestion'),
     },
     data: () => ({
       username: '',
       password: '',
+      usernameSuggestions: [],
+      suggestionTerm: '',
+      showDropdown: true,
+      highlightedIndex: -1,
     }),
+    watch: {
+      username: 'setSuggestionTerm',
+    },
     computed: {
       signUp() {
         return { name: PageNames.SIGN_UP };
       },
       versionMsg() {
         return this.$tr('poweredBy', { version: __version }); // eslint-disable-line no-undef
       },
+      canSignUp() {
+        return this.facilityConfig.learnerCanSignUp;
+      },
+      simpleLogin() {
+        return this.facilityConfig.learnerCanLoginWithNoPassword;
+      },
+      suggestions() {
+        // Filter suggestions on the client side so we don't hammer the server
+        return this.usernameSuggestions.filter(sug => sug.startsWith(this.username));
+      },
+      uniqueMatch() {
+        // If we have a matching username entered, don't show any suggestions.
+        return this.suggestions.length === 1 && this.suggestions[0] === this.username;
+      }
     },
     methods: {
+      handleKeyboardNav(e) {
+        if (this.showDropdown && this.suggestions.length) {
+          switch (e.code) {
+            case 'ArrowDown':
+              this.highlightedIndex = Math.min(
+                this.highlightedIndex + 1, this.suggestions.length - 1);
+              break;
+            case 'Enter':
+              this.fillUsername(this.suggestions[this.highlightedIndex]);
+              e.preventDefault();
+              break;
+            case 'Escape':
+              this.showDropdown = false;
+              break;
+            case 'ArrowUp':
+              this.highlightedIndex = Math.max(this.highlightedIndex - 1, -1);
+              break;
+            default:
+          }
```

Comment:
Totally a personal preference thing, but I dislike `switch` statements because of the `break`s, `default`, and weird indentation rules. Reminds me of coding in assembly :)

One alternative:

```javascript
handleKeyboardNav(e) {
  if (this.showDropdown && this.suggestions.length) {
    if (e.code === 'Enter') {
      this.fillUsername(this.suggestions[this.highlightedIndex]);
      e.preventDefault();
    } else if (e.code === 'Escape') {
      this.showDropdown = false;
    } else if (e.code === 'ArrowDown') {
      this.highlightedIndex = Math.min(this.highlightedIndex + 1, this.suggestions.length - 1);
    } else if (e.code === 'ArrowUp') {
      this.highlightedIndex = Math.max(this.highlightedIndex - 1, -1);
    }
  }
},
```